torchmultimodal/modules/encoders/clip_text_encoder.py

```diff
@@ -86,12 +86,13 @@ def initialize_parameters(self) -> None:
         nn.init.normal_(self.projection.weight, std=self.width**-0.5)
 
     def build_attention_mask(self) -> torch.Tensor:
-        mask = torch.full((self.context_length, self.context_length), True).triu(1)
+        mask = torch.full((self.context_length, self.context_length), int(True)).triu(1)
         return mask.to(device=None, dtype=torch.bool)
 
     def forward(self, text: torch.Tensor) -> torch.Tensor:
         embeddings = self.encoder(text, attn_mask=self.build_attention_mask())
         # [n_ctx, bs, transformer.width] -> [bs, n_ctx, transformer.width]
+        assert isinstance(embeddings, torch.Tensor)
```

Comment:
Why do we need the assertion here? Also, it's better to `if/raise` a specific error instead of a general assertion error.

Reply:
The assertion is needed by torchscript because aten::permute (on the next line) only takes tensors as input, but the encoder output is annotated as either a tensor or a list of tensors. I'll change the type check to the if/raise format.